client/src/components/utils/imageCrop.ts

```diff
@@ -0,0 +1,43 @@
+import { Area } from "react-easy-crop/types";
+
+const createImage = (url: string): Promise<HTMLImageElement> =>
+    new Promise((resolve, reject) => {
+        const image = new Image();
+        image.addEventListener("load", () => resolve(image));
+        image.addEventListener("error", (error) => reject(error));
+        image.src = url;
+        return image;
+    });
+
+export const getCroppedImg = async (imageSrc: string, pixelCrop: Area): Promise<string> => {
+    const image: HTMLImageElement = await createImage(imageSrc);
+    const canvas = document.createElement("canvas");
+    const ctx = canvas.getContext("2d");
+    const maxSize = Math.max(image.width, image.height);
+    const safeArea = 2 * ((maxSize / 2) * Math.sqrt(2));
```

Comment:
Bit weird, can just do maxSize * Math.sqrt(2) I think. What is here seems to be fine but logic is not really straightforward